### PR TITLE
User registration us12

### LINKED ISSUE #41: User Story 12, Registration Email must be unique
As a visitor
When I visit the user registration page
If I fill out the registration form
But include an email address already in the system
Then I am returned to the registration page
My details are not saved and I am not logged in
The form is filled in with all previous data except the email field and password fields
I see a flash message telling me the email address is already in use

### PR DESCRIPTION
**Pull request by leahriffell**
closes #41 

**Contributors**
 - Drivers: Priya, Leah 
 - Navigators: Luke, Grant

**What this did**
  - User registration throws error message if email is already in user table or confirmation password does not match
  - Refactored create action in user controller

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -3,17 +3,20 @@ class UsersController < ApplicationController
...
       redirect_to '/profile'
-    else
-      flash[:failure] = "You are missing one or more required fields"
-      redirect_to '/register'
+    elsif User.email_exists?(@user.email)
+      flash[:failure] = @user.errors.full_messages.to_sentence
+      @user.email = nil
+      render :new
+    else 
+      flash[:failure] = @user.errors.full_messages.to_sentence
```
Comment (GDemps) [inline app/controllers/users_controller.rb:17]: Great use of easier syntax.

File: app/views/users/_form.html.erb [hunk 2]
```diff
@@ -12,9 +12,9 @@
 <%= label_tag :email %><br>
 <%= text_field_tag :email, @user.email, placeholder: "Email" %><br>
 <%= label_tag :password %><br>
-<%= text_field_tag :password, @user.password, placeholder: "Password" %><br>
+<%= text_field_tag :password, nil, placeholder: "Password" %><br>
```
Comment (GDemps) [inline app/views/users/_form.html.erb:15]: Awesome, our passwords are working properly with BCrypt.

**Review by GDemps — APPROVED** [review]
Good refactor and functionality.

File: app/views/users/_form.html.erb (hunk 2 above)
Comment (leahriffell) [reply app/views/users/_form.html.erb:15]: Yes! It doesn't let a user save without confirmation pw matching.

**Pull request merged**